### ISSUE TITLE
Avoid synchronous operations implementation

### ISSUE THREAD

**Issue by amazuerar** [stefan-niedermann/nextcloud-notes#1318]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍  to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug** 

Dear Developer!

My name is Alejandro Mazuera-Rozo, I am a PhD Student at Universidad de los Andes, and at Università della Svizzera italiana. I am part of a research on the usage of network libraries within Android apps. As result of this we identified some code locations that might have network related problems.

In this case, we address the code locations that are related to a potential issue because of implementing synchronous operations. As you might know, using the `execute()` method on a call object will perform a synchronous request which is not recommended and its usage is discouraged.

In **synchronous operations**, tasks are performed one at a time and only when one is completed, the following is unblocked.

**Potential Code Location with synchronous operations:**

https://github.com/stefan-niedermann/nextcloud-notes/blob/3c95434b2080f6cd1500b54ee805a04596cb9e69/app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesServerSyncTask.java#L134

**To Reproduce**

Not applicable since this issue was found by performing a static analysis on source code

**Expected behavior**

With **asynchronous operations** one can move to another task before the previous one finishes, allowing you to deal with multiple requests simultaneously. Moreover, by avoiding this approach, it is less likely to perform network operations on the main thread. Android requires you to perform network operations on a thread other than the main UI thread; a `NetworkOnMainThreadException` is thrown otherwise.
       
In order to address this issue we recommend you to:

1. See: https://developer.android.com/training/basics/network-ops/connecting
2. Consider alternative methods (e.g., `enqueue()` form `Retrofit`) handling asynchronous operations

**Screenshots**

N/A

**Smartphone (please complete the following information):**
 - Nextcloud Notes-Version (android app):  N/A
 - F-Droid or Play Store:  N/A
 - Android-Version:  N/A
 - Device: N/A

**Server**

N/A

**Stacktrace**

N/A

**Comment by stefan-niedermann** [comment]
Please check the code again. The linked code fragment is called from an executor asynchronously. As you may know, Android does not even allow network operations on the UI thread and would throw an exception in such a case.
